src/coinbase/address/wallet_address.ts

```diff
@@ -322,22 +337,270 @@ export class WalletAddress extends Address {
   }
 
   /**
-   * Checks if trading is possible and raises an error if not.
+   * Checks if amount is valid and raises an error if not.
    *
    * @param amount - The amount of the Asset to send.
-   * @param fromAssetId - The ID of the Asset to trade from. For Ether, eth, gwei, and wei are supported.
+   * @param assetId - The ID of the Asset to trade from. For Ether, eth, gwei, and wei are supported.
    * @throws {Error} If the private key is not loaded, or if the asset IDs are unsupported, or if there are insufficient funds.
    */
-  private async validateCanTrade(amount: Amount, fromAssetId: string) {
+  private async validateAmount(amount: Amount, assetId: string) {
     if (!this.canSign()) {
       throw new Error("Cannot trade from address without private key loaded");
     }
-    const currentBalance = await this.getBalance(fromAssetId);
+    const currentBalance = await this.getBalance(assetId);
     amount = new Decimal(amount.toString());
     if (currentBalance.lessThan(amount)) {
       throw new Error(
         `Insufficient funds: ${amount} requested, but only ${currentBalance} available`,
       );
     }
   }
+
+  /**
+   * Creates a staking operation to stake, signs it, and broadcasts it on the blockchain.
+   *
+   * @param amount - The amount for the staking operation.
+   * @param assetId - The asset to the staking operation.
+   * @param action - The type of staking action to perform.
+   * @param timeoutSeconds - The amount to wait for the transaction to complete when broadcasted.
+   * @param intervalSeconds - The amount to check each time for a successful broadcast.
+   * @param options - Additional options such as setting the mode for the staking action.
+   * @returns The staking operation after it's completed fully.
+   */
+  public async createStakingOperation(
```

Comment:
any reason we don't expose a action specific method such as `createStakeOperation` similar to external address ?

Reply:
Yeah I'm planning on refactoring that a bit after our internal discussion. Will finalize things soon.